src/features/posts.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type PostsState = {
+  posts: Post[];
+  isLoading: boolean;
+  error: string;
+};
+
+const initialState: PostsState = {
+  posts: [],
+  isLoading: false,
+  error: '',
+};
+
+export const init = createAsyncThunk('posts/fetch', async (userId: number) => {
+  const posts = await getUserPosts(userId);
+
+  return posts;
+});
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    clearPosts: state => {
+      state.posts = [];
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(init.pending, state => {
+      state.isLoading = true;
+    });
+
+    builder.addCase(init.fulfilled, (state, action: PayloadAction<Post[]>) => {
+      state.posts = action.payload;
+      state.isLoading = false;
+    });
+
+    builder.addCase(init.rejected, state => {
+      state.error = 'Error';
```

Comment:
The 'init.rejected' case should ideally handle the error more specifically. Instead of setting a generic 'Error' message, consider using the error message from the rejected action payload to provide more detailed feedback.